client/network/src/protocol/generic_proto/handler/notif_out.rs

```diff
@@ -357,7 +368,10 @@ impl ProtocolsHandler for NotifsOutHandler {
 		match &mut self.state {
 			State::Open { substream, initial_message } =>
 				match Sink::poll_flush(Pin::new(substream), cx) {
-					Poll::Pending | Poll::Ready(Ok(())) => {},
+					Poll::Pending | Poll::Ready(Ok(())) =>
+						if let Some(metric) = &self.queue_size_report {
+							metric.observe(substream.queue_len() as f64);
```

Comment:
> One question is: should be really call observe() when we flush the queue? Should we not call it only when we push a message, so that the value represents the size that a queue had when we sent a message?

I don't think we should call `observe` when flushing the queue. From the top of my head I don't see how that would help.